docs/user_manual/components.md

```diff
@@ -1245,3 +1245,74 @@ node_1 --- transformer_4 --- node_2 --- line_5 --- node_3
 source_6     |                                    load_7
       transformer_tap_regulator_8
 ```
+
+### Voltage Regulator
+
+* type name: `voltage_regulator`
+* base: {hoverxreftooltip}`user_manual/components:regulator`
+
+`voltage_regulator` defines a regulator for voltage-controlled generators in the grid.
+A voltage regulator adjusts the reactive power output of a generator to maintain the voltage at its connection node at a specified setpoint.
+
+The voltage regulator changes the reactive power output of the generator it regulates to achieve the reference voltage `u_ref` at the generator's node.
+If `q_min` and `q_max` are provided, the reactive power is constrained within this range (i.e., `q_min <= q <= q_max` or `q_min >= q >= q_max`).
+If these limits are not provided, the reactive power can take any value needed to maintain the voltage setpoint.
+
+```{note}
+The `regulated_object` must reference a generator (`sym_gen` or `asym_gen`).
+Each generator can have at most one voltage regulator.
+When multiple voltage-regulated generators are connected to the same node, they should all specify the same `u_ref` value to avoid conflicting voltage setpoints.
+```
+
+```{warning}
+Reactive power limit checking is not yet fully implemented. When `q_min` and `q_max` are specified, the intended behavior is that if the required reactive power to maintain `u_ref` exceeds these limits, the voltage regulator should operate at the limit and the voltage may deviate from `u_ref`.
+```
+
+#### Input
+
+| name     | data type | unit                       | description                                         |           required           |  update  | valid values |
+| -------- | --------- | -------------------------- | --------------------------------------------------- | :--------------------------: | :------: | :----------: |
+| `u_ref`  | `double`  | -                          | reference voltage in per-unit at the generator node | &#10024; only for power flow | &#10004; |    `> 0`     |
+| `q_min`  | `double`  | volt-ampere-reactive (var) | minimum reactive power limit of the generator       | &#10060;                     | &#10004; |              |
+| `q_max`  | `double`  | volt-ampere-reactive (var) | maximum reactive power limit of the generator       | &#10060;                     | &#10004; |              |
+
+#### Steady state output
+
+| name              | data type         | unit                       | description                                                          |
+| ----------------- | ----------------- | -------------------------- | -------------------------------------------------------------------- |
+| `q`               | `RealValueOutput` | volt-ampere-reactive (var) | reactive power provided by the voltage regulator                     |
+| `limit_violated`  | `int8_t`          | -                          | reactive power limit violation indicator (not yet fully implemented) |
```

Comment:
unrelated to this PR but maybe we should add such a flag to the tap regulator as well. Relates to #1215